src/__tests__/helpers.ts

```diff
@@ -242,6 +256,56 @@ export async function makeConnection(agentA: Agent, agentB: Agent) {
   }
 }
 
+export async function makeTransport(
+  agent: Agent,
+  inboundTransporter: InboundTransporter,
+  outboundTransporter: OutboundTransporter
+) {
+  agent.setInboundTransporter(inboundTransporter)
+  agent.setOutboundTransporter(outboundTransporter)
+  await agent.init()
```

Comment:
yes, but inside of init,
```ts
    if (this.inboundTransporter) {
      await this.inboundTransporter.start(this)
    }
```
`start` on `inboundTransporter` is called. The interface requires an agent to be passed in when `start` is called so a server has access to receive messages.
for example.
```ts
  this.agent.receiveMessage(payload)
```
We chose to call `init` on the agent instead of calling start on the transport before setting it in the agent. 
When we add the ability to handle multiple types of transports I think we could improve on the interface.